cpp/src/traversal/bfs_kernels.cuh

```diff
@@ -564,7 +565,8 @@ __global__ void bottom_up_large_degree_kernel(IndexType *left_unvisited,
         if (lvl_u == (lvl - 1)) { valid_parent = u; }
       }
 
-      unsigned int warp_valid_p_ballot = cugraph::detail::utils::ballot((valid_parent != -1));
+      unsigned int warp_valid_p_ballot =
+        cugraph::detail::utils::ballot(valid_parent != ~IndexType(0));
```

Comment:
invalid_vertex_id<IndexType>::value is much more readable than ~IndexType(0)

Reply:
Done.